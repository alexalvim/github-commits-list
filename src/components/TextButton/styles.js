import styled from 'styled-components';

import Colors from '../../colors';
import Typo from '../../typo';

const Button = styled.button`
  background: none;
  border: none;
  color: ${Colors.mainColor};
  cursor: pointer;
  font-size: ${Typo.medium};

  ${(props) => props.wrapperStyle}
`;

export {
  Button
};