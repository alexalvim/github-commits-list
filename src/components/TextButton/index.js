import React from 'react';

import { Button } from './styles';

export default ({ label, onClick, wrapperStyle }) => (
  <Button
    wrapperStyle={wrapperStyle}
    onClick={onClick}>
    {label}
  </Button>
)