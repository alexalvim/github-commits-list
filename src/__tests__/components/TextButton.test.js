import React from 'react';
import { shallow } from 'enzyme';

import TextButton from '../../components/TextButton';

describe('Testing TextButton', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<TextButton label='test' />);

    expect(wrapper).toMatchSnapshot();
  });

  it('call expected function', () => {
    const mockFunction = jest.fn();
    const wrapper = shallow(<TextButton label='test' onClick={mockFunction} />);

    wrapper.simulate('click');
    expect(mockFunction).toHaveBeenCalled();    
  })
});