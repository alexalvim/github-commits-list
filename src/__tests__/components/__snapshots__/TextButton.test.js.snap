// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TextButton should render correctly 1`] = `ShallowWrapper {}`;
